### ISSUE TITLE
Incorrect Handling of HTTP codes (204 specifically)

### ISSUE THREAD

**Issue by mariari** [atlas-engineer/nyxt#1714]
As we have all seen, nyxt does not handle `history-backwards` very well. However it is lesser known that it does not handle direct `mp4` and `webms` urls.

From a discussion I've had with @jmercouris months ago it seems to be related to HTML codes, namely the part of the code that gets triggered is 

https://github.com/atlas-engineer/nyxt/blob/master/source/renderers/gtk.lisp#L985

I've referenced this in the comment below in issue #1677

See the comment there for an example mp4 that does not render

https://github.com/atlas-engineer/nyxt/issues/1677#issuecomment-894588175

According to IRC logs I have I first found this bug in an update around the time of `2021-04-23`.  

### Debug log 

This was taken by putting `(break)` on line 983 of the gtk file

from `history-backwars`

```lisp


Backtrace:
 0: ((LAMBDA (NYXT::WEB-VIEW NYXT::LOAD-EVENT NYXT::FAILING-URL ERROR) :IN FFI-BUFFER-MAKE) #<unused argument> #<unused argument> "https://duckduckgo.com/?q=test+wikipedia&ia=web" #<unused argument>)
      Locals:
        NYXT::FAILING-URL = "https://duckduckgo.com/?q=test+wikipedia&ia=web"
 1: (GOBJECT::CALL-WITH-RESTARTS #<FUNCTION (LAMBDA (NYXT::WEB-VIEW NYXT::LOAD-EVENT NYXT::FAILING-URL ERROR) :IN FFI-BUFFER-MAKE) {1006EDC4DB}> (#<CL-WEBKIT2:WEBKIT-WEB-VIEW {1006E370B3}> :WEBKIT-LOAD-CO..
      Locals:
        ARGS = (#<CL-WEBKIT2:WEBKIT-WEB-VIEW {1006E370B3}> :WEBKIT-LOAD-COMMITTED "https://duckduckgo.com/?q=test+wikipedia&ia=web" #S(CL-WEBKIT2::G-ERROR :DOMAIN 2115 :CODE 302 :MESSAGE "Load request cancelled"))
        FN = #<FUNCTION (LAMBDA (NYXT::WEB-VIEW NYXT::LOAD-EVENT NYXT::FAILING-URL ERROR) :IN FFI-BUFFER-MAKE) {1006EDC4DB}>
 2: ((LAMBDA (GOBJECT::CLOSURE GOBJECT::RETURN-VALUE GOBJECT::COUNT-OF-ARGS GOBJECT::ARGS GOBJECT::INVOCATION-HINT GOBJECT::MARSHAL-DATA) :IN "/home/katya/Documents/Workspace/Lisp/Projects/nyxt/_build/sub..
      Locals:
        GOBJECT::ARGS = #.(SB-SYS:INT-SAP #X7FB0F7DAD000)
        GOBJECT::CLOSURE = #.(SB-SYS:INT-SAP #X7FB0F03C82B0)
        GOBJECT::COUNT-OF-ARGS = 4
        GOBJECT::RETURN-VALUE = #.(SB-SYS:INT-SAP #X7FB0F7DACFA0)
 


#<CONS {1009A766A7}>
--------------------
A proper list:
0: #<CL-WEBKIT2:WEBKIT-WEB-VIEW {1006E370B3}>
1: :WEBKIT-LOAD-COMMITTED
2: "https://duckduckgo.com/?q=test+wikipedia&ia=web"
3: #S(CL-WEBKIT2::G-ERROR :DOMAIN 2115 :CODE 302 :MESSAGE "Load request cancelled")

```
This one seems to be an error in handling the code 302

From the mp4 error

```lisp

Backtrace:
 0: ((LAMBDA (NYXT::WEB-VIEW NYXT::LOAD-EVENT NYXT::FAILING-URL ERROR) :IN FFI-BUFFER-MAKE) #<unused argument> #<unused argument> "https://user-images.githubusercontent.com/39150985/107167284-c7af8580-69b..
      Locals:
        NYXT::FAILING-URL = "https://user-images.githubusercontent.com/39150985/107167284-c7af8580-69b8-11eb-834b-f7293b241c90.mp4"
 1: (GOBJECT::CALL-WITH-RESTARTS #<FUNCTION (LAMBDA (NYXT::WEB-VIEW NYXT::LOAD-EVENT NYXT::FAILING-URL ERROR) :IN FFI-BUFFER-MAKE) {100BE77BEB}> (#<CL-WEBKIT2:WEBKIT-WEB-VIEW {100702F303}> :WEBKIT-LOAD-CO..
      Locals:
        ARGS = (#<CL-WEBKIT2:WEBKIT-WEB-VIEW {100702F303}> :WEBKIT-LOAD-COMMITTED "https://user-images.githubusercontent.com/39150985/107167284-c7af8580-69b8-11eb-834b-f7293b241c90.mp4" ..)
        FN = #<FUNCTION (LAMBDA (NYXT::WEB-VIEW NYXT::LOAD-EVENT NYXT::FAILING-URL ERROR) :IN FFI-BUFFER-MAKE) {100BE77BEB}>
 2: ((LAMBDA (GOBJECT::CLOSURE GOBJECT::RETURN-VALUE GOBJECT::COUNT-OF-ARGS GOBJECT::ARGS GOBJECT::INVOCATION-HINT GOBJECT::MARSHAL-DATA) :IN "/home/katya/Documents/Workspace/Lisp/Projects/nyxt/_build/sub..
      Locals:
        GOBJECT::ARGS = #.(SB-SYS:INT-SAP #X7FB0F7DAD000)
        GOBJECT::CLOSURE = #.(SB-SYS:INT-SAP #X7FB0F03C82B0)
        GOBJECT::COUNT-OF-ARGS = 4
        GOBJECT::RETURN-VALUE = #.(SB-SYS:INT-SAP #X7FB0F7DACFA0)


#<COMPILED-FUNCTION {53CBC25B}>
--------------------
NAME: (LAMBDA (NYXT::WEB-VIEW NYXT::LOAD-EVENT NYXT::FAILING-URL ERROR) :IN FFI-BUFFER-MAKE)
ARGLIST: (NYXT::WEB-VIEW NYXT::LOAD-EVENT NYXT::FAILING-URL ERROR)
TYPE: (FUNCTION (T T T T) *)
CODE: #<code id=13BDA [247] (SB-PCL::FAST-METHOD FFI-BUFFER-MAKE (GTK-BUFFER)), "DEFINE-FFI-METHOD FFI-BUFFER-MAKE" {53CB4C7F}>
DOCUMENTATION: NIL


#<CONS {10068338C7}>
--------------------
A proper list:
0: #<CL-WEBKIT2:WEBKIT-WEB-VIEW {100702F303}>
1: :WEBKIT-LOAD-COMMITTED
2: "https://user-images.githubusercontent.com/39150985/107167284-c7af8580-69b8-11eb-834b-f7293b241c90.mp4"
3: #S(CL-WEBKIT2::G-ERROR :DOMAIN 2230 :CODE 204 :MESSAGE "Plug-in handled load")


```

This one seems like a code 204 error

**Comment by Ambrevar** [comment]
@aartaka Would you have the time to look into this?

**Comment by aartaka** [comment]
> @aartaka Would you have the time to look into this?

Sounds fun, I'm in!

**Comment by Ambrevar** [comment]
I knew you'd find it entertaining! :D

**Comment by aartaka** [comment]
Should be fixed by a3f6516614c055cd28f360df580406a9bb7aa6b8.

@mariari, thanks for your thorough investigation. It wasn't HTTP code, though. `204` was WebKitGTK error code for `WEBKIT_PLUGIN_ERROR_WILL_HANDLE_LOAD` which they somewhy were emitting when the plugin was taking the responsibility for page loading (as it happens with GStreamer and .mp4, I guess).  So, we're handling all the necessary HTTP status codes, phew 😅 

Anyway, closing this, as videos seem to work just fine on a3f6516614c055cd28f360df580406a9bb7aa6b8.
